File: examples/rest_quote.py
#!/usr/bin/env python3
"""Request one public AssetFare quote without MCP, auth, or execution."""

from __future__ import annotations

import json
import math
import sys
import urllib.request


# USD 1 is reachability/schema smoke only. The representative default is USD
# 1,000 native USDC; always compare fresh alternatives at the intended amount.
amount_usd = float(sys.argv[1]) if len(sys.argv) > 1 else 1000
from_chain = sys.argv[2].lower() if len(sys.argv) > 2 else "solana"
from_token = sys.argv[3].upper() if len(sys.argv) > 3 else "USDC"
to_chain = sys.argv[4].lower() if len(sys.argv) > 4 else "base"
to_token = sys.argv[5].upper() if len(sys.argv) > 5 else "USDC"
if not math.isfinite(amount_usd) or amount_usd < 1:
    raise SystemExit("amount must be a finite USD number of at least 1")

payload = json.dumps(
    {
        "from_chain": from_chain,
        "from_token": from_token,
        "to_chain": to_chain,
        "to_token": to_token,
        "amount_usd": amount_usd,
    },
    separators=(",", ":"),
).encode()
request = urllib.request.Request(
    "https://api.assetfare.dev/v2/quote",
    data=payload,
    headers={
        "content-type": "application/json",
        "user-agent": "AssetFareRestQuoteExample/1",
    },
    method="POST",
)
with urllib.request.urlopen(request, timeout=15) as response:
    print(json.dumps(json.load(response), indent=2, sort_keys=True))

File: examples/rest-quote.mjs
const endpoint = "https://api.assetfare.dev/v2/quote";
const amountUsd = Number(process.argv[2] || 1);
const fromChain = String(process.argv[3] || "solana").toLowerCase();
const fromToken = String(process.argv[4] || "SOL").toUpperCase();
const toChain = String(process.argv[5] || "base").toLowerCase();
const toToken = String(process.argv[6] || "USDC").toUpperCase();

if (!Number.isFinite(amountUsd) || amountUsd < 1) {
  throw new Error("amount must be a finite USD number of at least 1");
}

const response = await fetch(endpoint, {
  method: "POST",
  headers: {
    "content-type": "application/json",
    "user-agent": "AssetFareRestQuoteExample/1",
  },
  body: JSON.stringify({
    from_chain: fromChain,
    from_token: fromToken,
    to_chain: toChain,
    to_token: toToken,
    amount_usd: amountUsd,
  }),
  signal: AbortSignal.timeout(15_000),
});

const body = await response.json();
if (!response.ok) {
  throw new Error(`quote failed: HTTP ${response.status} ${JSON.stringify(body)}`);
}

console.log(JSON.stringify(body, null, 2));
